### ISSUE TITLE
Gradient Accumulation

### ISSUE THREAD

**Issue by ojaffe** [ojaffe/Music-Performance#4]
Implement [gradient accumulation](https://towardsdatascience.com/what-is-gradient-accumulation-in-deep-learning-ec034122cfa), since this is very memory intensive.
- Allow for variable batch size
- Test difference in memory usage
- Test difference in training / inference speed
